### PR TITLE
feat: added ticket_sequence to Transaction class common fields

### PR DESCRIPTION
**Pull request by connorjchen**
## High Level Overview of Change

Added ticket_sequence field to common Transaction class fields

### Context of Change

Based on issue "Add common field ticket_sequence to Transaction #413" 

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan

Wrote out unit tests

### CODE REVIEW

**Comment by mvadari** [comment]
nit: this PR should have a name somewhat akin to https://www.conventionalcommits.org/en/v1.0.0/ (see: other PRs)

File: tests/unit/models/transactions/test_transaction.py [hunk 1]
```diff
@@ -79,3 +80,44 @@ def test_to_dict_flag_list(self):
...
+        tx = Transaction(
+            account=_ACCOUNT,
+            fee=_FEE,
+            ticket_sequence=_TICKET_SEQUENCE,
+            transaction_type=TransactionType.ACCOUNT_DELETE,
+        )
+        value = tx.to_dict()["ticket_sequence"]
+        self.assertEqual(value, _TICKET_SEQUENCE)
+
+    def test_to_dict_ticket_sequence_with_sequence_zero(self):
```
Comment (JST5000) [inline tests/unit/models/transactions/test_transaction.py:94]: Basic intent with this suggestion is to make it really obvious to future readers exactly what this test should do, especially if they have no context on why this test exists. 
```suggestion
    def test_throws_when_ticket_sequence_and_sequence_both_non_zero(self):
```

File: tests/unit/models/transactions/test_transaction.py [hunk 2]
```diff
@@ -79,3 +80,44 @@ def test_to_dict_flag_list(self):
...
+            account=_ACCOUNT,
+            fee=_FEE,
+            sequence=0,
+            ticket_sequence=_TICKET_SEQUENCE,
+            transaction_type=TransactionType.ACCOUNT_DELETE,
+        )
+        value = tx.to_dict()["ticket_sequence"]
+        self.assertEqual(value, _TICKET_SEQUENCE)
+
+    def test_ticket_sequence_with_sequence_nonzero(self):
```
Comment (JST5000) [inline tests/unit/models/transactions/test_transaction.py:105]: Basic intent with this suggestion is to make it really obvious to future readers exactly what this test should do, especially if they have no context on why this test exists. 
```suggestion
    def test_throws_when_ticket_sequence_and_sequence_both_non_zero(self):
```

Comment (JST5000) [reply tests/unit/models/transactions/test_transaction.py:105]: (Initially made the comment on the wrong test case)

File: tests/unit/models/transactions/test_transaction.py [hunk 3]
```diff
@@ -79,3 +80,44 @@ def test_to_dict_flag_list(self):
...
+        with self.assertRaises(XRPLModelException):
+            Transaction(
+                account=_ACCOUNT,
+                fee=_FEE,
+                sequence=_SEQUENCE,
+                ticket_sequence=_TICKET_SEQUENCE,
+                transaction_type=TransactionType.ACCOUNT_DELETE,
+            )
+
+    def test_ticket_sequence_with_account_txn_id(self):
```
Comment (JST5000) [inline tests/unit/models/transactions/test_transaction.py:115]: ditto - this is a bit vague about what the expected outcome should be.

File: xrpl/models/transactions/transaction.py [hunk 4]
```diff
@@ -320,12 +320,24 @@ class Transaction(BaseModel):
...
+
     txn_signature: Optional[str] = None
     """
     The cryptographic signature from the sender that authorizes this
     transaction. Automatically added during signing.
     """
 
+    def _get_errors(self: Transaction) -> Dict[str, str]:
+        errors = super()._get_errors()
+        if(self.ticket_sequence is not None and (self.sequence is not None and self.sequence != 0 or self.account_txn_id is not None)):
```
Comment (JST5000) [inline xrpl/models/transactions/transaction.py:337]: I would add an extra pair of parenthesis just to really clarify the order of operations here.
```suggestion
        if(self.ticket_sequence is not None and ((self.sequence is not None and self.sequence != 0) or self.account_txn_id is not None)):
```

File: xrpl/models/transactions/transaction.py [hunk 5]
```diff
@@ -320,12 +320,24 @@ class Transaction(BaseModel):
...
+            errors["Transaction"] = "If ticket sequence provided, account_txn_id must be None and sequence must be None or 0"
```
Comment (JST5000) [inline xrpl/models/transactions/transaction.py:338]: nit to keep things consistent
```suggestion
            errors["Transaction"] = "If ticket_sequence provided, account_txn_id must be None and sequence must be None or 0"
```

**Review by JST5000 — APPROVED** [review]
LGTM

**Review by ckniffen — APPROVED** [review]

**Review by JST5000 — CHANGES_REQUESTED** [review]
Oh, could you actually also add this as an entry in HISTORY.md?

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Function to parse the final account balances from a transaction's metadata
 - Function to parse order book changes from a transaction's metadata
 - Support for Ed25519 seeds that don't use the `sEd` prefix
+- Common field ticket_sequence to Transaction class
```
Comment (intelliot) [inline CHANGELOG.md:13]:
```suggestion
- Common field `ticket_sequence` to Transaction class
```

File: xrpl/models/transactions/transaction.py [hunk 7]
```diff
@@ -243,12 +243,30 @@ class Transaction(BaseModel):
     added during signing.
     """
 
+    ticket_sequence: Optional[int] = None
+    """
+    The sequence number of the ticket to use in place of a Sequence number. If
+    this is provided, Sequence must be 0. Cannot be used with account_txn_id.
```
Comment (intelliot) [inline xrpl/models/transactions/transaction.py:249]:
```suggestion
    this is provided, sequence must be 0. Cannot be used with account_txn_id.
```

**Review by intelliot — APPROVED** [review]

**Comment by intelliot** [comment]
Maybe consider adding a link to https://xrpl.org/tickets.html somewhere in here? Just a thought

**Pull request merged**